usd/src/sdf_parser/Geometry.cc

```diff
@@ -204,6 +204,36 @@ namespace usd
       fullName =
         ignition::common::findFile(uri.Str());
     }
+    else if (uri.Scheme() == "model")
+    {
+      fullName =
+        ignition::common::findFile(_geometry.MeshShape()->Uri());
+
+      std::function<void(const std::string)> addSubFolders =
+        [&](const std::string &_dir)
+      {
+        for (ignition::common::DirIter file(_dir);
+          file != ignition::common::DirIter(); ++file)
+        {
+          std::string current(*file);
+          if (ignition::common::isDirectory(current))
+          {
+            auto systemPaths = ignition::common::systemPaths();
+            systemPaths->AddFilePaths(current);
+            addSubFolders(current);
```

Comment:
I don't get why this needs to be recursive. My understanding was that you add the directories that are siblings of the model directory so that a model can reference meshes in other models, but this shouldn't require recursing to subdirectories of the model.